p2pserver/message/utils/msg_handler.go

```diff
@@ -642,3 +642,31 @@ func GetHeadersFromHash(startHash common.Uint256, stopHash common.Uint256) ([]*t
 
 	return headers, nil
 }
+
+// EmergencyRequestHandle forwards emergency request msg to emergency governance module to handle
+func EmergencyRequestHandle(data *msgTypes.MsgPayload, p2p p2p.P2P, pid *evtActor.PID, args ...interface{}) {
+	log.Debug("RX emergency request message", data.Addr, data.Id)
+
+	if pid != nil {
+		msg := data.Payload.(*msgTypes.EmergencyReqMsg)
+		input := &msgCommon.EmergencyMsg{
+			MsgType: msgCommon.EmergencyReq,
+			Content: &msg.Payload,
+		}
+		pid.Tell(input)
```

Comment:
I think we should verify the EmergencyRequest here? (signatures in msg)

Reply:
In my opinion, verify the EmergencyRequest is handled by emergency goverance module. Here message handler just route it to the correct module.